apps/labs/posts/code-generator-and-nebari.md

```diff
@@ -0,0 +1,159 @@
+
+Some of the related PRs can be found here:[#314](https://github.com/pytorch-ignite/code-generator/pull/314) [#265](https://github.com/pytorch-ignite/code-generator/pull/265)
+
+### 3. The changes in the CI and other Vuejs based parts of the code base (mainly based on JS)
+
+I had to contribute some JavaScript-based code to the app, but since I had not worked on a JS project before, I needed to learn the basics before contributing to these issues, especially [Vue.js](https://vuejs.org) :-( . So, I tried to learn by reading some online tutorials and documentation. I particularly liked this [tutorial](https://www.youtube.com/watch?v=YrxBCBibVo0&list=PL4cUxeGkcC9hYYGbV60Vq3IXYNfDk8At1) series on YouTube.
+
+To explain more about how the generates templates, let's assume I made some files like `main.py`, `model.py`, and `utils.py` as a template. Now we use the [ejs](https://ejs.co) project to design the templates. To illustrate this, let us take an example from the templates.
+
+```python
+#::: if ((it.argparser == 'fire')) { :::#
+    fire.Fire(main)
+#::: } else if ((it.argparser == 'hydra')){ :::#
+    sys.argv.append("hydra.run.dir=.")
+    sys.argv.append("hydra.output_subdir=null")
+    sys.argv.append("hydra/job_logging=stdout")
+    main()
+#::: } else { :::#
+    main()
+#::: } :::#
+```
+
+Now, as you can see, there are some commented codes. This is part of JS, which helps select different configurations of [argparsers](http://argparsers.It). Seems easy, right?
+
+But it can be challenging to manage, and we also need to ensure we satisfy the `lint` formatting for the CI (which can be pretty tricky with templates :-( ). Still, these JS features can be very powerful, and we are trying to improve these as much as possible. By the way, if you are confused about where this `it.argparser` selector came from, better check the project, but the short answer is it comes from a `metadata.json` file that maintains the options for all the templates in the app.
+
+Some of the related PRs can be found here: [#288](https://github.com/pytorch-ignite/code-generator/pull/288) [#283](https://github.com/pytorch-ignite/code-generator/pull/283)
+
+# The Way Forward for the Project
+
+The Code generator project seems to be moving forward at a great pace and is expected to have many more additions in the future. Starting and maintaining a new project is challenging in open source and can have a very high risk/reward ratio. Still, good communities try their best to make and maintain projects that can be helpful for the maximum number of people and reduce the friction for new people entering the community. Some of the issues I suggest and may work on in the future are
```

Comment:
Again, just to make sure the text flows better and the punctuation is respected.

```suggestion
The Code generator project seems to be moving forward at a great pace and is expected to have many more additions in the future. Starting and maintaining a new project is challenging in open source and can have a very high risk/reward ratio. Still, good communities try their best to make and maintain projects that can be helpful for the maximum number of people and reduce the friction for new people entering the community. Here are some of the issues I suggest and may work on in the future.
```